app/src/main/java/com/android/sample/ui/profile/Profile.kt

```diff
@@ -474,3 +345,333 @@ fun InterestBox(interest: String) {
         Text(text = interest, fontSize = SUBTITLE_FONTSIZE.sp, color = Color.Black)
       }
 }
+
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun UserProfile(
+    user: User,
+    navigationActions: NavigationActions,
+    imageViewModel: ImageViewModel,
+    profileViewModel: ProfileViewModel,
+    listActivitiesViewModel: ListActivitiesViewModel,
+    uid: String
+) {
+  var activityType by remember { mutableIntStateOf(0) }
+  val uiState by listActivitiesViewModel.uiState.collectAsState()
+  val activitiesList = (uiState as ListActivitiesViewModel.ActivitiesUiState.Success).activities
+  val userActivities =
+      activitiesList.filter {
+        it.creator == user.id || it.participants.map { it.id }.contains(user.id)
+      }
+  val hourDateViewModel = HourDateViewModel()
+  var showMenu by remember { mutableStateOf(false) }
+  val context = LocalContext.current
+  val networkManager = NetworkManager(context)
+
+  Scaffold(
+      bottomBar = {
+        if (uid == "") {
+          BottomNavigationMenu(
+              onTabSelect = { route -> navigationActions.navigateTo(route) },
+              tabList = LIST_TOP_LEVEL_DESTINATION,
+              selectedItem = navigationActions.currentRoute())
+        }
+      },
+      topBar = {
+        TopAppBar(
+            modifier = Modifier.testTag("profileTopBar"),
+            title = {
+              Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+                Text(
+                    modifier = Modifier.testTag("userName"),
+                    text = user.name,
+                    style =
+                        TextStyle(
+                            fontSize = MEDIUM_PADDING.sp,
+                            fontWeight = FontWeight(MAXIMUM_FONT_WEIGHT),
+                            color = Color(DARK_GRAY),
+                        ))
+              }
+            },
+            navigationIcon = {
+              if (uid != "") {
+                GoBackButton(navigationActions)
+              }
+            },
+            actions = {
+              if (uid == "") {
+                IconButton(
+                    onClick = { showMenu = true },
+                    modifier = Modifier.testTag("moreOptionsButton")) {
+                      Icon(
+                          imageVector = Icons.Default.Settings,
+                          contentDescription = "More options",
+                          modifier = Modifier.testTag("settingsIcon"))
+                    }
+                DropdownMenu(expanded = showMenu, onDismissRequest = { showMenu = false }) {
+                  DropdownMenuItem(
+                      modifier = Modifier.testTag("logoutMenuItem"),
+                      text = { Text("Logout") },
+                      onClick = {
+                        performOfflineAwareAction(
```

Comment:
Maybe you can put this in a try catch block in case of potential network issues

Reply:
Thank you for the suggestion! The `performOfflineAwareAction` function already handles potential network issues, so adding an extra `try-catch` block might be redundant. However, I’ll review the code to ensure proper error handling is in place and make improvements if necessary.

Reply:
Understood! thanks :)